src/streams.h

```diff
@@ -296,6 +296,16 @@ class CDataStream
         data.insert(data.end(), begin(), end());
         clear();
     }
+
+    /**
+     * XOR the contents of this stream with a certain key.
+     *
+     * @param[in] key    The key used to XOR the data in this stream.
+     */
+    void Xor(const std::string& key) {
```

Comment:
Really good suggestion, will integrate. Two questions:
- won't a `key` that is `"\000..."` have `.size() == 0`? should we just return early if that's the case instead of blowing up with an `assert`?
- I put this in `CDataStream` because, basically, it has to mutate `vch`. That's a protected member variable, so a method `Xor` became. Is there an alternate way to do this that you'd prefer?